hid.pyx

```diff
@@ -124,6 +148,28 @@ cdef class device:
             hid_close(self._c_hid)
             self._c_hid = NULL
 
+    def __enter__(self):
+        """Opening statement for `with ...` clause.
+
+        Either `open...()` method must be called in the same line!
+
+        The `with ...` context manager will make sure the object is opened and closed
+        (regardless of error status).
+
+        Specify either `path` or the other values. See :func:`~hid.device.open` and
+        :func:`~hid.open_path` for usages.
+        """
+        if self._c_hid == NULL:
+            raise IOError('Device not open, make sure to call it like: '
```

Comment:
This raises an `IOError`, but other similar assertions (e.g. in `write`) raise `ValueError`.